### DISCUSSION TITLE
Installing Windows Terminal on Non-Persistent VDI with No Internet

### DISCUSSION THREAD

**Discussion by DHarwell** [microsoft/terminal#13407] [General]
Hello Everyone,

We are currently using VMWare Horizon to host non-persistent Windows 10 virtual machines. These VMs have no internet access, so I'm downloading the MSIXBundle for Windows Terminal and attempting to install it with the Add-AppxPackage PowerShell command. Unfortunately, whether I use a logon task or try to run the command by hand, it appears to complete but Terminal doesn't show up on the start menu at all. Am I even going about this the right way? If not, is there some way we can just install Terminal for all users in the template we're building the virtual machines from so we can skip the logon task entirely?

Thanks for any advice.
-Derek

**Comment by DHowett** [comment]
Hi! That's an excellent question.

You definitely can install Terminal while building the template, _if_ you can use DISM (or the DISM cmdlets) on the source image.

You'll want to grab one the _Preinstallation Kits_ available for stable builds on [our releases page](https://github.com/microsoft/terminal/releases) [1] and follow the guide linked from the top of the release notes:

> A [preinstallation](https://docs.microsoft.com/en-us/windows/msix/desktop/deploy-preinstalled-apps) kit is available for system integrators and OEMs interested in prepackaging Windows Terminal with a Windows image. More information is available in the [DISM documentation on preinstallation](https://docs.microsoft.com/windows-hardware/manufacture/desktop/preinstall-apps-using-dism). Users who do not intend to preinstall Windows Terminal should continue using the msixbundle distribution.

If you can't customize the source image with DISM, you may still be able to install Terminal: you'll just need to grab the preinstall kit and make certain to install all of the architecture-appropriate `appx`, `msix` and `msixbundle` files inside of it using `Add-AppxPackage`. Just installing the Terminal one is no longer sufficient (since we have some dependencies!)

[1]
<img width="673" alt="image" src="https://user-images.githubusercontent.com/189190/176768364-a99500ba-0519-4f99-a226-649aa8628c1a.png">
(sorry, wow, that's ugly)

**Reply by DHarwell** [reply]
Thank you for your help, Unfortunately, Horizon works by deploying from a Snapshot of the source VM, so I won't be able to use DISM. However, the Preinstall you helped me find has gotten me most of the way there. I'm now able to install the right version of the VCLibs as well as the Terminal, but Terminal still doesn't show up in the start menu list. If I pull up the store and go to My Library > Installed, I can see it listed there but launching it just gives me a Task Bar Icon with no actual interactable program. 

Get-AppxPackage seems to indicate it's installed as well. Any ideas what I might be missing?

```
Name              : Microsoft.WindowsTerminal
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X64
ResourceId        :
Version           : 1.13.11431.0
PackageFullName   : Microsoft.WindowsTerminal_1.13.11431.0_x64__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.WindowsTerminal_1.13.11431.0_x64__8wekyb3d8bbwe
IsFramework       : False
PackageFamilyName : Microsoft.WindowsTerminal_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
Dependencies      : {Microsoft.VCLibs.140.00.UWPDesktop_14.0.30704.0_x64__8wekyb3d8bbwe}
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok

Name              : Microsoft.VCLibs.140.00.UWPDesktop
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X64
ResourceId        :
Version           : 14.0.30704.0
PackageFullName   : Microsoft.VCLibs.140.00.UWPDesktop_14.0.30704.0_x64__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.VCLibs.140.00.UWPDesktop_14.0.30704.0_x64__8wekyb3d8bbwe
IsFramework       : True
PackageFamilyName : Microsoft.VCLibs.140.00.UWPDesktop_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok
```

**Comment by WSLUser** [comment]
This is related to the msix platform issue. See https://github.com/microsoft/terminal/issues/6010#issuecomment-792800655

**Comment by SpyassL** [comment]
try this to install Windows Teminal:
```ps1
winget install --id=Microsoft.WindowsTerminal -e
```
